### PR TITLE
vstart.sh: fix CEPH_PORT check and cleanups

### PR DESCRIPTION
**Pull request by iotcg**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/vstart.sh [hunk 1]
```diff
@@ -748,21 +748,23 @@ EOF
...
-	    echo "{\"cephx_secret\": \"$OSD_SECRET\"}" > $CEPH_DEV_DIR/osd$osd/new.json
-            ceph_adm osd new $uuid -i $CEPH_DEV_DIR/osd$osd/new.json
-	    rm $CEPH_DEV_DIR/osd$osd/new.json
-            $SUDO $CEPH_BIN/ceph-osd -i $osd $ARGS --mkfs --key $OSD_SECRET --osd-uuid $uuid
-
-            local key_fn=$CEPH_DEV_DIR/osd$osd/keyring
-	    cat > $key_fn<<EOF
-[osd.$osd]
-	key = $OSD_SECRET
+        local key_fn=$CEPH_DEV_DIR/osd$osd/keyring
```
Comment (tchaikov) [inline src/vstart.sh]: @iotcg could you be more specific in the commit message? "refine" is kind of vague in this context. how about 

> add newly created osd's auth before mkfs to avoid following warning message:

?

File: src/vstart.sh [hunk 2]
```diff
@@ -746,6 +746,7 @@ EOF
 		ln -s $kstore_path $CEPH_DEV_DIR/osd$osd
 	    else
 		mkdir -p $CEPH_DEV_DIR/osd$osd
+		touch $CEPH_DEV_DIR/osd$osd/block
```
Comment (tchaikov) [inline src/vstart.sh]: @iotcg how is this related to the error message of

> auth: unable to find a keyring on dev/osd0/keyring: (2) No such file or directory

?

File: src/vstart.sh [hunk 3]
```diff
@@ -921,7 +921,7 @@ then
   while [ true ]
   do
     CEPH_PORT="$(echo $(( RANDOM % 1000 + 40000 )))"
-    ss -a -n | egrep ":${CEPH_PORT} .+LISTEN" 1>/dev/null 2>&1 || break
+    ss -a -n | egrep ":${CEPH_PORT}\>" 2>/dev/null | egrep "\<LISTEN\>" -q || break
```
[the remark this answers is not in the document]
Comment (iotcg) [reply src/vstart.sh]: @tchaikov What does ":25" standard for? The port number is not fixed.

[the remark this answers is not in the document]
Comment (tchaikov) [reply src/vstart.sh]: sorry, it should have been
```
ss -a -n | egrep "\<LISTEN\>.+:${CEPH_PORT}\s+" 1>/dev/null 2>&1 || break
```

[the remark this answers is not in the document]
Comment (iotcg) [reply src/vstart.sh]: Get it. I'll follow your suggestion to change it.

File: src/vstart.sh [hunk 4]
```diff
@@ -28,7 +28,7 @@ if [ -e CMakeCache.txt ]; then
   CEPH_ROOT=$(get_cmake_variable ceph_SOURCE_DIR)
   CEPH_BUILD_DIR=`pwd`
   [ -z "$MGR_PYTHON_PATH" ] && MGR_PYTHON_PATH=$CEPH_ROOT/src/pybind/mgr
-  CEPH_MGR_PY_VERSION_MAJOR=$(get_cmake_variable MGR_PYTHON_VERSION | cut -d '.' -f1)
+  CEPH_MGR_PY_VERSION_MAJOR=$(get_cmake_variable MGR_PYTHON_VERSION | cut -d '.' -f 1)
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/vstart.sh]: @iotcg this comment is not addressed. also please note, "1" is not a "slice" in the terminology of "cut", it is a **f**ield.

File: src/vstart.sh [hunk 5]
```diff
@@ -941,7 +941,7 @@ then
   while [ true ]
   do
     CEPH_PORT="$(echo $(( RANDOM % 1000 + 40000 )))"
-    ss -a -n | egrep ":${CEPH_PORT} .+LISTEN" 1>/dev/null 2>&1 || break
+    ss -a -n | egrep "\<LISTEN\>.+:${CEPH_PORT}\s+" 1>/dev/null 2>&1 || break
```
Comment (tchaikov) [inline src/vstart.sh]: nit, s/ramdon/random/ in the commit message.

File: src/vstart.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (iotcg) [reply src/vstart.sh]: Thanks kefu

File: src/vstart.sh (hunk 5 above)
Comment (tchaikov) [reply src/vstart.sh]: we are close. please s/randon/random/.

Comment (iotcg) [reply src/vstart.sh]: Done
